src/lib.rs

```diff
@@ -17,6 +17,7 @@ mod test;
 pub struct Heap {
     bottom: usize,
     size: usize,
+    max_size: usize,
```

Comment:
For what do we need `max_size`? The only reason I can see is that this makes it possible to make `extend` safe. Maybe it's better to have an `unsafe` extend function and let the caller manage `max_size` itself.

(Maybe we should even get rid of the `bottom` and `size` fields… Then we could expand the heap with an arbitrary memory chunk without the need for a single continuous block.)